fragment.go

```diff
@@ -480,6 +513,13 @@ func (f *fragment) unprotectedSetBit(rowID, columnID uint64) (changed bool, err
 func (f *fragment) clearBit(rowID, columnID uint64) (bool, error) {
 	f.mu.Lock()
 	defer f.mu.Unlock()
+	err, mustClose := f.reopen()
```

Comment:
this pattern is bugging me, but i can't see a good way around it. you can't have a function which does a defer-for-a-parent, but this feels like something where if we want to change this logic, we have to change it in too many places. i note that, in nearly all cases, it also follows a mutex lock/unlock pair. maybe something like

`err, defer := f.assureFile(mustLock)`, then if err is non-nil, we return it as an error. and after that check, if defer is non-nil, we call it. but... hmm. this still feels clunky. i was going to say "what if that all went into a wrapper", but of course, these ARE the wrappers around the `unprotected` methods.

i have not started a "review" because i'm sleepy-headed and want to read this again tomorrow.

Reply:
I had to change it several times while developing and it was clunky....

for what it's worth, I have some refactoring in my private TODO list which will greatly reduce the number of places we need to do this. I think setBit, clearBit, and others can be trivially rewritten to use `importPositions`